ipam/utils.go

```diff
@@ -56,24 +55,24 @@ func (e *NotFoundError) Error() string {
 	return "Object not found"
 }
 
-func updateObject(cl client.Client, ctx context.Context, obj runtime.Object, opts ...client.UpdateOption) error {
-	err := cl.Update(ctx, obj.DeepCopyObject(), opts...)
+func updateObject(cl client.Client, ctx context.Context, obj client.Object, opts ...client.UpdateOption) error {
+	err := cl.Update(ctx, obj.DeepCopyObject().(client.Object), opts...)
```

Comment:
casting an interface like this normally return 2 values, and you should check the second one to verify the casting happened properly.